messaging-activemq/subsystem/src/main/java/org/wildfly/extension/messaging/activemq/ServerAdd.java

```diff
@@ -316,14 +319,14 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
 
             // Add the ActiveMQ Service
             ServiceName activeMQServiceName = MessagingServices.getActiveMQServiceName(serverName);
-            final ServiceBuilder serviceBuilder = serviceTarget.addService(activeMQServiceName);
-            Supplier pathManager = serviceBuilder.requires(context.getCapabilityServiceName(PATH_MANAGER_CAPABILITY, PathManager.class));
+            final CapabilityServiceBuilder serviceBuilder = capabilityServiceTarget.addCapability(ACTIVEMQ_SERVER_CAPABILITY);
+            Consumer<ActiveMQServer> consumer = serviceBuilder.provides(activeMQServiceName);
```

Comment:
Pass ACTIVEMQ_SERVER_CAPABILITY to provides as well. That way the consumer is used for injection into dependents for both service names.